lib/config/config.js

```diff
@@ -40,7 +43,7 @@ function loadConfigFile(configPath) {
 		return require(configPath);
 	}
 
-	if (extension === '.json') {
+	if (extension === '.json' || extension === '.json5') {
```

Comment:
One issue here: in case of syntax errors, the linter code below will not work properly against JSON5 files.